Tellduslive

molobrakos opened the pull request:
Added rudimentary support for the Telldus Live service, used with the Tellstick Net device.
http://www.telldus.se/products/tellstick_net
http://live.telldus.com/

Commit:
initial version of yr.no weather component

Commit:
Updated yr sensor

Commit:
Added requirements to yr sensor

Commit:
Tests for yr sensor

Commit:
Fix in yr sensor

Commit:
Added units to sensor yr

Commit:
try to fix requirements for yr sensor

Commit:
Update yr.py

Commit:
Update yr.py

Commit:
Update yr.py

Commit:
added comment for yr sensor

Commit:
fix tests in yr sensor

Commit:
Refactor yr sensor

Commit:
Refactor yr sensor

balloob committed:
Merge pull request #680 from balloob/dev

Version 0.9.0

balloob committed:
Merge pull request #706 from balloob/dev

0.9.1

Commit:
skeleton initial commit

Commit:
skeleton initial commit

Commit:
added tellive-py

Commit:
added tellive-py

Commit:
added tellduslive

Commit:
added tellduslive

fabaff committed:
Equalize log messages

fabaff committed:
Remove space

Commit:
initial support for tellduslive/tellstick net

Commit:
initial support for tellduslive/tellstick net

Commit:
more work on support for tellduslive

Commit:
more work on support for tellduslive

Commit:
tellduslive sensors work

Commit:
tellduslive sensors work

Commit:
support for switches

Commit:
support for switches

Commit:
support for tellduslive/tellstick net

Commit:
support for tellduslive/tellstick net

Commit:
lint

Commit:
lint

balloob committed:
Merge pull request #778 from balloob/dev

0.10.0.rc1

balloob committed:
Merge branch 'dev'

Commit:
Added yr sensor to requirements_all

Commit:
changed to use requestes in stead of urllib for yr sensor

Commit:
changed to use requestes in stead of urllib for yr sensor

Commit:
small fix in yr sensor name

Commit:
small fix in yr sensor name

Commit:
Merge remote-tracking branch 'upstream/dev' into tellduslive

philipbl committed:
Rename geofancy to locative (fix #761)

molobrakos committed:
extracted constant

molobrakos committed:
extracted constant

molobrakos committed:
only add devices, not groups

molobrakos committed:
only add devices, not groups

molobrakos committed:
Merge branch 'tellduslive' of githubid:molobrakos/home-assistant into tellduslive

balloob committed:
Fix Alexa bug if no value for slots

balloob committed:
Fix EntityComponent deadlock

balloob committed:
Merge branch 'release-101' into dev

molobrakos committed:
one decimal for temperature, zero decimals for humidity

molobrakos committed:
one decimal for temperature, zero decimals for humidity

molobrakos committed:
lint

molobrakos committed:
lint

andythigpen committed:
Reset log handlers to lowest level.

This is necessary to enable logging lower than INFO for the error log file.

molobrakos committed:
add icon

molobrakos committed:
add icon

molobrakos committed:
icon depending on sensor type

molobrakos committed:
icon depending on sensor type

molobrakos committed:
lint

molobrakos committed:
lint

molobrakos committed:
added telllive

molobrakos committed:
added telllive

fabaff committed:
Enable logging-too-many-args

Commit:
Logbook entry events are now lower case

balloob committed:
Merge pull request #790 from andythigpen/logger-fix

Reset log handlers to lowest level.

Commit:
Set default log level to INFO

balloob commented on homeassistant/components/sensor/eliqonline.py:
```diff
@@ -63,6 +63,10 @@ def state(self):
         """ Returns the state of the device. """
         return self._state
 
+    @property
```
This is an unrelated change, please remove.

molobrakos replied:
Thanks, will do

Commit:
Merge branch 'yr_no' into dev

Commit:
Merge branch 'pr/780' into dev

balloob committed:
Merge pull request #786 from philipbl/locative

Rename geofancy to locative (fix #761)

balloob commented on homeassistant/components/tellduslive.py:
```diff
@@ -0,0 +1,139 @@
+from datetime import timedelta
+import logging
+
+from tellcore.constants import (
```
You cannot reference any of your requirements outside of the methods. As the file will be loaded before the requirements are installed.

balloob commented on homeassistant/components/tellduslive.py:
```diff
@@ -0,0 +1,139 @@
+        return False
+
+    global NETWORK
+    NETWORK = TelldusLiveData(config)
```
Can you validate here that the keys are valid?

balloob commented on homeassistant/components/tellduslive.py:
```diff
@@ -0,0 +1,139 @@
+    NETWORK = TelldusLiveData(config)
+
+    for component_name, discovery_type in (
+            ('switch', DISCOVER_SWITCHES),
```
Can you make sure that you only fire the discovery event if there are actually sensors available?

balloob commented on homeassistant/components/switch/tellduslive.py:
```diff
@@ -0,0 +1,77 @@
+    @property
+    def should_poll(self):
+        """ Tells Home Assistant to poll this entity. """
+        return True
```
There doesn't seem to be an update method here so there is no need to poll.

molobrakos replied:
The switches needs to be reworked I realize, and the underlying library (telllive) currently seems to miss the required functionality. Would you prefer me closing the pull request until then?

balloob replied:
If the switch turning on/off works you can just have it return `False` for now and keep the optimistic approach of predicting state.

balloob commented on homeassistant/components/switch/tellduslive.py:
```diff
@@ -0,0 +1,77 @@
+
+_LOGGER = logging.getLogger(__name__)
+DEPENDENCIES = ['tellduslive']
+TELLSTICK_UNKNOWN = 0
```
This doesn't seem to be used.

balloob commented on homeassistant/components/switch/tellduslive.py:
```diff
@@ -0,0 +1,77 @@
+"""
+import logging
+
+from tellcore.constants import (
```
Have to be imported inside a method.

balloob commented on .coveragerc:
```diff
@@ -15,6 +15,7 @@ omit =
     homeassistant/components/*/modbus.py
 
     homeassistant/components/*/tellstick.py
+    homeassistant/components/*/tellduslive.py
```
Also add the main component to this file.

andythigpen committed:
Set the root logger to lowest level in logger component.

In combination with resetting the log level on the handlers, this
allows messages lower than the default INFO to be logged when using
the logger component.

balloob committed:
Merge pull request #791 from andythigpen/logger-fix

Set the root logger to lowest level in logger component.

philipbl committed:
Fix bug in rest sensor

olavhaugen commented:
There should be support for "light" also just like Tellstick.

Commit:
Add functionality set random color Philips Hue

Commit:
Fix styling issues

molobrakos committed:
removed unrelated fix

molobrakos committed:
removed unrelated fix

molobrakos committed:
import requirements inside methods

molobrakos committed:
import requirements inside methods

molobrakos committed:
added icon

Commit:
Merge remote-tracking branch 'upstream/dev' into tellduslive

persandstrom committed:
changed to python-verisure 0.4.1

persandstrom committed:
log texts

persandstrom committed:
revert accidental change

persandstrom committed:
added docstring

molobrakos committed:
icons

molobrakos committed:
icons

Commit:
Merge remote-tracking branch 'origin/eliq-icon' into tellduslive

Commit:
Merge remote-tracking branch 'origin/systemmon-icons' into tellduslive

molobrakos committed:
Merge branch 'tellduslive' of githubid:molobrakos/home-assistant into tellduslive

molobrakos committed:
oops, reverted merge

molobrakos committed:
oops, reverted merge

philipbl committed:
Simplify error handling

molobrakos committed:
oops

molobrakos committed:
oops

molobrakos committed:
added main component

molobrakos committed:
added main component

balloob committed:
Merge pull request #793 from philipbl/rest_bug_fix

Fix bug in rest sensor

balloob committed:
Merge pull request #798 from R1chardTM/hue-random

Add random color effect for Philips Hue

balloob committed:
Merge pull request #797 from molobrakos/eliq-icon

replace default icon

balloob committed:
Merge pull request #796 from persandstrom/verisure-update

Verisure update

molobrakos committed:
only discover available componenents. throttle only sensors, not switches.

molobrakos committed:
only discover available componenents. throttle only sensors, not switches.

molobrakos committed:
fewer member variables

molobrakos committed:
fewer member variables

molobrakos committed:
moved dependency on external constants

molobrakos committed:
moved dependency on external constants

molobrakos committed:
removed debug

molobrakos committed:
removed debug

balloob committed:
Travis: Less verbose requirements_all.txt check

molobrakos committed:
lint + cache sensor values, not switches

molobrakos committed:
lint + cache sensor values, not switches

Commit:
Merge branch 'pr/792' into dev

balloob commented on homeassistant/components/sensor/eliqonline.py:
```diff
@@ -54,11 +54,6 @@ def name(self):
         return self._name
 
     @property
-    def icon(self):
```
I've already merged your other PR to add this icon so now this PR is removing that icon again…

balloob replied:
You probably want to revert some of your changes and rebase on top of `dev`.

molobrakos committed:
get battery level and last update time as well

molobrakos committed:
get battery level and last update time as well

molobrakos committed:
get battery level and last update time as well

molobrakos committed:
get battery level and last update time as well

molobrakos committed:
Revert "removed unrelated fix"

This reverts commit cccda69d391a4627654e06befe9f6bcba8d3c0ef.

molobrakos committed:
Revert "removed unrelated fix"

This reverts commit cccda69d391a4627654e06befe9f6bcba8d3c0ef.

molobrakos committed:
Merge branch 'tellduslive' of githubid:molobrakos/home-assistant into tellduslive

molobrakos committed:
finally in sync?

molobrakos commented:
messed things up, will provide a new pull request on a clean branch, sorry
